File: Measurement Studio/Examples/DotNET/v4.5/UI.WinForms/Graph/CustomAnnotations/cs/CustomArrowTailStyle.cs
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using NationalInstruments.UI;

namespace NationalInstruments.Examples.CustomAnnotations
{
	public class FeatherTailStyle : NationalInstruments.UI.ArrowStyle
	{
		private const int numFeathers = 6;

        protected override PointF GetArrowLinePoint(PointF point, Size size)
		{
			return new PointF (point.X + size.Width, point.Y);
		}

		protected override void Draw(object context, NationalInstruments.UI.ArrowStyleDrawArgs args)
		{
			Graphics graphics = args.Graphics;
			Color color = args.Color;
			Size size = args.Size;
			PointF point = args.Point;
			float lineWidth = args.LineWidth;

			using(GraphicsPath path = new GraphicsPath())
			{
				for(int i=0; i<numFeathers; i++)
				{
					GetFeatherPolygon(path, point, size, i);
				}

				using(Pen pen = new Pen(color, lineWidth))
				{
					graphics.DrawPath(pen, path);
				}
			}  
		}
              
		private static void GetFeatherPolygon(GraphicsPath path, PointF point, Size size, int pos)
		{
			float dist = Convert.ToSingle(size.Width) / Convert.ToSingle(numFeathers);
			float offset = pos * dist;

			path.AddLine(point.X + offset, point.Y, point.X + offset + dist, point.Y);
			path.AddLine(point.X + offset + dist, point.Y, point.X + offset, point.Y - size.Height/2);
			path.AddLine(point.X + offset, point.Y - size.Height/2, point.X + offset + dist, point.Y);
			path.AddLine(point.X + offset + dist, point.Y, point.X + offset, point.Y + size.Height/2);
			path.AddLine(point.X + offset, point.Y + size.Height/2, point.X + offset + dist, point.Y);
		}
	}
}

File: Measurement Studio/Examples/DotNET/v4.5/UI.WinForms/Graph/CustomAnnotations/vb/CustomArrowTailStyle.vb
Imports System.Drawing.Drawing2D

Public Class FeatherTailStyle
    Inherits ArrowStyle
    Private Const numFeathers As Integer = 6

    Protected Overloads Overrides Sub Draw(ByVal context As Object, ByVal args As UI.ArrowStyleDrawArgs)
        Dim graphics As graphics = args.Graphics
        Dim color As color = args.Color
        Dim size As size = args.Size
        Dim point As PointF = args.Point
        Dim lineWidth As Single = args.LineWidth

        Dim path As GraphicsPath = New GraphicsPath
        Try
            For i As Integer = 0 To numFeathers
                GetFeatherPolygon(path, point, size, i)
            Next i

            Dim pen As pen = New pen(color, lineWidth)
            graphics.DrawPath(pen, path)
            pen.Dispose()

        Finally
            path.Dispose()
        End Try
				

    End Sub

    Protected Overrides Function GetArrowLinePoint(ByVal point As System.Drawing.PointF, ByVal size As System.Drawing.Size) As System.Drawing.PointF
        Return New PointF(point.X + size.Width, point.Y)
    End Function

    Private Shared Sub GetFeatherPolygon(ByRef path As GraphicsPath, ByVal point As PointF, ByVal size As Size, ByVal pos As Integer)

        Dim dist As Single = Convert.ToSingle(size.Width) / Convert.ToSingle(numFeathers)
        Dim offset As Single = pos * dist

        path.AddLine(point.X + offset, point.Y, point.X + offset + dist, point.Y)
        path.AddLine(point.X + offset + dist, point.Y, point.X + offset, Convert.ToSingle(point.Y - size.Height / 2.0))
        path.AddLine(point.X + offset, Convert.ToSingle(point.Y - size.Height / 2), point.X + offset + dist, point.Y)
        path.AddLine(point.X + offset + dist, point.Y, point.X + offset, Convert.ToSingle(point.Y + size.Height / 2))
        path.AddLine(point.X + offset, Convert.ToSingle(point.Y + size.Height / 2), point.X + offset + dist, point.Y)
    End Sub
End Class
